### ISSUE TITLE
Double connection

### ISSUE THREAD

**Issue by tdwesten** [rodgc/ngx-socket-io#77]
Hi there, I got a double connection when using this code. Is there a way to prevent this? 

```
import { Injectable } from '@angular/core';
import { Socket } from 'ngx-socket-io';
import { AuthService } from '../auth/auth.service';
import { WSMessage } from '../dto/i-ws-message';

@Injectable({
  providedIn: 'root',
})
export class WebsocketService {
  constructor(private socket: Socket, private _auth: AuthService) {}

  receiveUpdates(channel: string) {
    return this.socket.fromEvent(channel);
  }
}
```

NestJS backend log: 
```
[Nest] 59975   - 2020-11-07 12:24:00   [AppGateway] Client disconnected: _BLVWByuUyrDezIQAAAC
[Nest] 59975   - 2020-11-07 12:24:00   [AppGateway] Client disconnected: _BLVWByuUyrDezIQAAAC
[Nest] 59975   - 2020-11-07 12:24:01   [AppGateway] Client connected: NaKNu7dpQlh7kuDcAAAD
[Nest] 59975   - 2020-11-07 12:24:01   [AppGateway] Client connected: NaKNu7dpQlh7kuDcAAAD
[Nest] 59975   - 2020-11-07 12:27:06   [AppGateway] Client disconnected: NaKNu7dpQlh7kuDcAAAD
[Nest] 59975   - 2020-11-07 12:27:06   [AppGateway] Client disconnected: NaKNu7dpQlh7kuDcAAAD
[Nest] 59975   - 2020-11-07 12:27:07   [AppGateway] Client connected: aC1HCY4emBuYp0X8AAAE
[Nest] 59975   - 2020-11-07 12:27:07   [AppGateway] Client connected: aC1HCY4emBuYp0X8AAAE
```

**Comment by DanielDanninger** [comment]
you found a solutio? I run in the same issue.

**Comment by tdwesten** [comment]
@DanielDanninger Nah, no luck so far...

**Comment by Ruslancic** [comment]
Same here +1!

**Comment by Helveg** [comment]
Does upgrading to v3.3.1 help?

**Comment by Ruslancic** [comment]
> Does upgrading to v3.3.1 help?

Upgrading is too much difficult : [https://github.com/rodgc/ngx-socket-io/issues/90](url)

**Comment by grafixen** [comment]
Same here +1
Tried using the latest version 4.0 on a vanilla cleanly generated Angular project. 
Getting multiple connections at once. 

Nestjs Example: https://github.com/nestjs/nest/tree/master/sample/02-gateways
Angular: Newly generated project with this in app.module.ts

`  imports: [
    BrowserModule,
    AppRoutingModule,
    SocketIoModule.forRoot({
      url: 'http://localhost:8081',
      options: {},
    }),
  ],
`

**Comment by Helveg** [comment]
@grafixen It looks like that sample is using a Socket.IO version 2.1? Version 4.0 requires the use of Socket.IO 4.0

**Comment by grafixen** [comment]
> @grafixen It looks like that sample is using a Socket.IO version 2.1? Version 4.0 requires the use of Socket.IO 4.0

You are right. Downgrading ngx-socket-io to v3.2.0 seems to solve the issue.

**Comment by MarceloPetrucio** [comment]
Thanks, it worked by changing the version to v3.2.0

**Comment by Alex-Cauble** [comment]
I was using the multiple connections method and calling super, after switching to the app.module  SocketIoModue.forRoot(config) use pattern it stopped emitting multiple events for things.

**Comment by Helveg** [comment]
People who used socket.io 4.0 wanted an ngx-socket-io that is compatible with that; This was first provided in v3.3.0, but since it was a breaking change this was reverted in version 3.3.1 and instead released as version 4.0

The semantic versioning is now respected, we ship the matching version of our peer dependency socket.io and a table on the README indicates version compatibility. You can now specify the major version that matches your server side socket.io and everything will work as expected:

* For socket.io 2-3: `"ngx-socket-io": "^3.3.1"
* For socket.io 4: `"ngx-socket-io": "4.x"

**Comment by MincDev** [comment]
I am still getting the double connection issue. I've tried installing the versions suggested, but this does not help.
